subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/DefaultTaskExecutionContext.java

```diff
@@ -52,13 +61,35 @@ public void setBuildCacheKey(TaskOutputCachingBuildCacheKey buildCacheKey) {
     }
 
     @Override
-    public UniqueId getOriginBuildInvocationId() {
-        return originBuildInvocationId;
+    public OriginTaskExecutionMetadata getOriginExecutionMetadata() {
+        return originExecutionMetadata;
     }
 
     @Override
-    public void setOriginBuildInvocationId(@Nullable UniqueId originBuildInvocationId) {
-        this.originBuildInvocationId = originBuildInvocationId;
+    public void setOriginExecutionMetadata(OriginTaskExecutionMetadata originExecutionMetadata) {
+        this.originExecutionMetadata = originExecutionMetadata;
+    }
+
+    @Override
+    public Timer getExecutionTimer() {
```

Comment:
I don't understand why we need to expose the `executionTimer`. It only is used to change the time shown in a logging statement. I think it would be better to not expose this timer.
WDYT?

Reply:
I don’t think yet another definition of time is better.

Reply:
I don't think the clock in `SkipCachedTaskExecuter` is about a definition of time but about measuring a certain time duration. It is the same use case as the timer in `SkipUpToDateTaskExecuter`, which is used to measure the time the up-to-date check took. https://github.com/gradle/gradle/blob/0533c676c7e73990536d9b548cc86a883d929773/subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/SkipUpToDateTaskExecuter.java#L47-L50

The build scan plugin uses build operation times to get those durations.

Maybe we can also remove the log statement completely.